jobs/bump-jenkins-plugins.Jenkinsfile

```diff
@@ -0,0 +1,142 @@
+                git clone --depth=1 --branch ${pr_branch} https://github.com/${fork_repo}.git
+                cd fedora-coreos-pipeline
+                git remote add upstream https://github.com/${repo}.git
+                git fetch upstream
+                git reset --hard upstream/main
+            """)
+
+            /* Read the plugins from the lockfile */
+            pluginslist = shwrapCapture("grep -v ^# ${plugins_lockfile}").split('\n')
+        }
+
+        stage("Check for plugin updates") {
+            def pluginUrl
+            def pluginsUpdateList = []
+            pluginslist.each { plugin ->
+                def parts = plugin.split(':')
+                if (parts.size() != 2) {
+                    error("Invalid plugin format: ${plugin}")
+                } else {
+                    def pluginName = parts[0]
+                    def currentVersion = parts[1]
+
+                    /* Retrieve the download URL for the most recent version of a Jenkins plugin from the Jenkins update center. 
+                       After following all redirects, curl prints the final URL to stdout. The final URL is captured in the pluginUrl variable for further use. */
+                    pluginUrl = shwrapCapture("curl -Ls -I -f -o /dev/null -w '%{url_effective}' https://updates.jenkins.io/download/plugins/${pluginName}/latest/${pluginName}.hpi")
+
+                    def latestVersion = getVersionFromPluginUrl(pluginUrl)
+                    if (latestVersion.toString() != currentVersion.toString()) {
+
+                        /* Update the plugin version in the lockfile */
+                        pluginsToUpdate["${pluginName}"] = [currentVersion, latestVersion]
+                        println("Plugin: ${pluginName} current version is ${currentVersion}, it will be updated to latest version: ${latestVersion}")
+
+                        /* Plugins to be updated are added to the Plugins Update List */
+                        pluginsUpdateList.add("-e s/${pluginName}:${currentVersion}/${pluginName}:${latestVersion}/g")
+
+                    } else {
+                        println("The latest version of ${pluginName} is already installed: ${currentVersion}")
+                    }
+                }
+            }
+            /* Plugins find/replace operation */
+            if (!pluginsUpdateList.isEmpty()) {
+                def pluginUpdate = "sed -i " + pluginsUpdateList.join(' ')
+                shwrap("""
+                    cd fedora-coreos-pipeline
+                    ${pluginUpdate} ${plugins_lockfile}
+                """)
+            }
+        }
+
+        /* Open a PR if there are plugin updates */
+        stage("Open a PR") {
+            if (shwrap("git diff --exit-code") != 0){
+                def message = "jenkins/plugins: update to latest versions"
+                shwrap("""
+                    cd fedora-coreos-pipeline
+                    git status
+                    git diff
```

Comment:
these two are just for the logs?

Reply:
Yes, I forgot to remove them . Had used it as part of debugging. will remove them